src/sqlfluff/core/rules/config_info.py

```diff
@@ -60,10 +60,18 @@
         "definition": "The capitalisation policy to enforce.",
     },
     "extended_capitalisation_policy": {
-        "validation": ["consistent", "upper", "lower", "pascal", "capitalise", "snake"],
+        "validation": [
+            "consistent",
+            "upper",
+            "lower",
+            "pascal",
+            "capitalise",
+            "snake",
+            "camel",
+        ],
         "definition": (
-            "The capitalisation policy to enforce, extended with PascalCase "
-            "and snake_case. "
+            "The capitalisation policy to enforce, extended with PascalCase, "
+            "snake_case, and camelCase. "
             "This is separate from ``capitalisation_policy`` as it should not be "
```

Comment:
I think the precedence rules here aren't very easy to guess. I think from running through the examples below, that I understand the logic - but I think we should document it here.

If nothing else - someone will ask on the slack channel and I'd like to be able to point them to something to explain _why_ the default config is acting the way that it is.